### PR TITLE
Stage/log failed component registrations

### PR DESCRIPTION
**Pull request by santiagonoguez**
## Description
Log error if there are unregistered components in a template after 60 seconds running.

Note that I had to add new unrelated test coverage so that it doesn't fail.

## Motivation and Context
Validate how widespread are templates where rise elements are not being registered.

## How Has This Been Tested?
Tested manually with this schedule with a template that has an invalid `<rise-not-valid>` element: https://apps.risevision.com/schedules/details/43cc8c2a-ffc7-47ab-a2c5-5efa40b1ee86?cid=7fa5ee92-7deb-450b-a8d5-e5ed648c575f

and validated using query and display id:

```
SELECT timestamp, eventErrorCode, eventDetails, debugInfo
FROM `endpoint-event-logs.logs.eventLog`
WHERE DATE(timestamp) = "2021-05-13"
AND endpointId = '35ERAQE7DCDC'
ORDER BY timestamp DESC
LIMIT 1000
```

returns

```
{
    "timestamp": "2021-05-13 16:25:57.625865 UTC",
    "eventErrorCode": "E999999999",
    "eventDetails": "{\"event\":\"rise elements not registered as web components\",\"eventDetails\":{\"details\":\"[\\\"rise-not-valid\\\"]\",\"errorCode\":\"E999999999\"}}",
    "debugInfo": "{\"template\":{\"product_code\":\"a4ef37a0486832bcc7d087a9b6dead1ea41a4f18\",\"version\":\"1.0.0\",\"name\":\"example-video-component\",\"presentation_id\":\"c7458a6f-dfc7-46be-a1ab-377430aae1c4\"}}"
  }
```

 - [x] Common template has been tested on Player
 - [x] Common template has been tested in Apps Template Editor

## Release Plan:
To be released before Friday

- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed.
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No need to update notification or notify support

### CODE REVIEW

**Comment by santiagonoguez** [comment]
oh coveralls stil failing :O damn !

**Comment by santiagonoguez** [comment]
oh, it's related to logging right ? because master is also failing :O :O :O

**Comment by stulees** [comment]
@santiagoarriaga I don't follow, what do you mean "master is also failing"?

**Comment by santiagonoguez** [comment]
Master coveralls failed: https://coveralls.io/builds/39626354

**Comment by stulees** [comment]
@santiagoarriaga This was noted in this PR - https://github.com/Rise-Vision/common-template/pull/203#issuecomment-839811689 given we did not remove the code base for client side logging but skipped the tests.

File: test/unit/rise-helpers.test.js [hunk 1]
```diff
@@ -592,6 +619,20 @@ describe( "Helpers", function() {
...
+    it( "should log if there are unregistered rise elements", function() {
+      var clock = _sandbox.useFakeTimers();
+
+      _sandbox.stub( RisePlayerConfiguration.Logger, "error" );
+
+      RisePlayerConfiguration.Helpers.startValidateRiseElementsAreRegistered();
+
+      clock.tick( 60000 );
+
+      expect( RisePlayerConfiguration.Logger.error ).to.have.been.calledOnce;
```
Comment (olegrise) [inline test/unit/rise-helpers.test.js:632]: @santiagonoguez can you clarify why `RisePlayerConfiguration.Logger.error` should have been called? For what element you expect error to be logged?

File: src/rise-helpers.js [hunk 2]
```diff
@@ -316,13 +323,33 @@ RisePlayerConfiguration.Helpers = (() => {
     _riseElements = null;
   }
 
+  function startValidateRiseElementsAreRegistered() {
+    setTimeout(() => {
+      const notRegistered = RisePlayerConfiguration.Helpers.getRiseElementsNotRegisteredAsWebComponents();
+
+      if ( notRegistered.length ) {
+        RisePlayerConfiguration.Logger.error({
+          name: "RisePlayerConfiguration",
```
Comment (stulees) [inline src/rise-helpers.js:332]: @santiagoarriaga Could you use `RisePlayerConfiguration. RISE_PLAYER_CONFIGURATION_DATA` here instead?

Comment (santiagonoguez) [reply src/rise-helpers.js:332]: @stulees I'll do, thanks !

File: test/unit/rise-helpers.test.js (hunk 1 above)
Comment (santiagonoguez) [reply test/unit/rise-helpers.test.js:632]: @olegrise for all 8, actually, look at test in line 296 above. The list of element is mocked, and none of them is registered with customElements :D

File: src/rise-helpers.js [hunk 3]
```diff
@@ -360,3 +388,7 @@ RisePlayerConfiguration.Helpers = (() => {
   return exposedFunctions;
 
 })();
+
+if ( !RisePlayerConfiguration.Helpers.isTestEnvironment()) {
+  RisePlayerConfiguration.Helpers.startValidateRiseElementsAreRegistered();
+}
```
Comment (olegrise) [inline src/rise-helpers.js:394]: @santiagonoguez just wondering about your choice to call `startValidateRiseElementsAreRegistered` outside of the RisePlayerConfiguration.Helpers object. Any specific reason for doing it that way?

Comment (santiagonoguez) [reply src/rise-helpers.js:394]: @olegrise I was just following what other files do, for example:
https://github.com/Rise-Vision/common-template/blob/master/src/rise-play-until-done.js#L92

File: test/unit/rise-helpers.test.js (hunk 1 above)
Comment (olegrise) [reply test/unit/rise-helpers.test.js:632]: @santiagoarriaga got it. Thanks.

File: src/rise-helpers.js (hunk 3 above)
Comment (olegrise) [reply src/rise-helpers.js:394]: okay, thanks.

**Review by olegrise — APPROVED** [review]

File: src/rise-helpers.js (hunk 2 above)
Comment (santiagonoguez) [reply src/rise-helpers.js]: @stulees updated !

**Review by stulees — APPROVED** [review]

**Pull request merged**